src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1211,9 +1211,21 @@ public static string GetModulePath(string currentProcessModulePath, string hklmM
                 // sharedModulePath
                 // systemModulePath
                 currentProcessModulePath = AddToPath(currentProcessModulePath, personalModulePathToUse, 0);
-                int insertIndex = PathContainsSubstring(currentProcessModulePath, personalModulePathToUse) + personalModulePathToUse.Length + 1;
+                int insertIndex = currentProcessModulePath.IndexOf(Path.PathSeparator, PathContainsSubstring(currentProcessModulePath, personalModulePathToUse));
```

Comment:
The logic looks still pretty fragile. For example:
1. When `personalModulePathToUse` contains multiple paths separated by `;`, `AddToPath(currentProcessModulePath, personalModulePathToUse, 0)` basically will reverse the order of paths in `personalModulePathToUse` when inserting at index 0. Then when the code flow reaches here, `PathContainsSubstring(currentProcessModulePath, personalModulePathToUse)` will return -1 because the path order was reversed in `AddToPath`;

1. When `personalModulePathToUse` contains `;;` in it, then `PathContainsSubstring(currentProcessModulePath, personalModulePathToUse)` is likely to return -1.

In those 2 cases (maybe more), this will fail with `ArgumentOutOfRangeException`.

Reply:
How about updating `AddToPath`, to make it return the next index to insert the next module path?

Reply:
`personalModulePathToUse` is a single path so it can't contain `;;` in it nor be multiple paths unless someone deliberately put that into `powershell.config.json`